src/components/SortDropdown.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+
+const useStyles = makeStyles((theme) => ({
+  hiddenStyle: {
+    display: 'none',
+  },
+  selectedOptionTextStyle: {
+    fontWeight: '500',
+    fontFamily: 'Work Sans',
+    fontSize: '14px',
+    lineHeight: '18px',
+    margin: 'auto 0 auto 0',
+  },
+  optionTextStyle: {
+    fontWeight: '500',
+    fontFamily: 'Work Sans',
+    fontSize: '14px',
+    lineHeight: '18px',
+    margin: 'auto 0 auto 10px',
+  },
+  dropdown: {
+    display: 'flex',
+    cursor: 'pointer',
+    backgroundColor: theme.palette.background.default,
+    boxSizing: 'border-box',
+    borderRadius: '4px 4px 0px 0px',
+    minHeight: '43px',
+    minWidth: '177px',
+    border: '1px solid #BCBCBC',
+  },
+  selectedOptionGridStyle: {
+    display: 'inline-flex',
+    marginLeft: '10px',
+  },
+  dropdownIconStyle: {
+    marginLeft: '10px',
+    wdith: '32px',
+    height: '29px',
+  },
+  iconGridStyle: {
+    margin: 'auto',
+    paddingTop: '6px',
+  },
+  optionContainerStyle: {
+    display: 'flex',
+    minHeight: '43px',
+    minWidth: '177px',
+    backgroundColor: theme.palette.background.default,
+    border: '0.5px solid #BCBCBC',
+    cursor: 'pointer',
+    '&:hover': {
+      backgroundColor: theme.palette.secondary.light,
+    },
+  },
+  optionsGridStyle: {
+    position: 'absolute',
+    borderRadius: '0px 0px 4px 4px',
+    left: '0',
+    right: '0',
+  },
+  dropdownContainer: {
+    position: 'relative',
+    display: 'inline-block',
+  },
+}));
+
+export default function SortDropdown({
+  inputSortMethodList,
+  defaultSortMethod,
+  setSortMethod,
+}) {
+  const classes = useStyles();
+  const [isDropdownExpand, setIsDropdownExpand] = useState(false);
+  const sortMethodList = inputSortMethodList;
+  const [selectedSortMethod, setSelectedSortMethod] =
+    useState(defaultSortMethod);
+  const [sortMethodMap] = useState({
+    'best match': 'Best Match',
+    updated: 'Last Updated',
+    stars: 'Stargazer Count',
+  });
```

Comment:
Hey Travis, so this was just a thought I had after reading through the commits; and please feel free to correct me at any point since I totally may be off with my observations. 

So, it looks like this component is rendering a list of sorting options in the form of a dropdown. There are some sorting methods here in state of this component (sortMethodMap), but I believe the component also receives a list of sort methods as a prop. 

If this is the case, is there a reason behind this approach? I was wondering if there are two sources for a list of sorting methods, then every time we want to add, remove, or change a sort method, both the prop and this state variable will have to be modified.

Reply:
I agrre. I change the useState to map const value to represent the mapping.